src/config.py

```
@@ -102,13 +102,15 @@ def __init__(self) -> None:
         self.config: RootConfig = RootConfig()
         self.load_config()
 
-    def deep_merge(self, base: dict, overrides: dict) -> dict:
-        """Recursive merge for dictionaries."""
+    def deep_merge(self, base: dict, overrides: dict, replace_keys: set[str] = None) -> dict:
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)